### ISSUE TITLE
possible slip on cudnn version?

### ISSUE THREAD

**Issue by astooke** [pytorch/pytorch#20844]
## 🐛 Bug

I think I may have just had a problem related to cudnn version.  Did a `conda install` with ` cudatoolkit=9.0`; CPU-only code was working (simple conv+mlp), GPU code was running but failing to learn; `torch.backends.cudnn.version()` returned `7501`, but I had `7003` in my `/usr/local/cuda/` (and cuda 9.0 installed).  Since upgrading that to `7501`, GPU code is learning fine.  :)

Nothing in my `.bashrc` was pointing to any specific cuda directory, but from the environment printout below, it's linking to my pre-installed cudnn,  despite the conda install supposedly including everything?  (Wish I had seen your `collect_env.py` previously!)

This was confusing and took a long time to figure out--maybe would be helpful to have an extra check in there somewhere on cudnn version?

Thanks!
-Adam

## Environment

PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.3 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce GTX 1080
GPU 1: GeForce GTX 1080

Nvidia driver version: 384.130
cuDNN version: /usr/local/cuda-9.0/targets/x86_64-linux/lib/libcudnn.so.7

Versions of relevant libraries:
[pip] numpy==1.16.3
[pip] torch==1.1.0
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.3                      199  
[conda] mkl_fft                   1.0.12           py37ha843d7b_0  
[conda] mkl_random                1.0.2            py37hd81dba3_0  
[conda] pytorch                   1.1.0           py3.7_cuda9.0.176_cudnn7.5.1_0    pytorch

**Comment by soumith** [comment]
@astooke we ship with our own CuDNN, it's statically linked to the pytorch binary, precisely because anaconda comes with an older version.

> GPU code was running but failing to learn

This does not sound right, our library does not even go looking for libcudnn.so (we dont link against it dynamically). Is there any way I might be able to reproduce it -- actually can you reproduce it again with that previous version in /usr/local/cuda? I was hoping you can run some `lsof` or `strace` commands and see who is loading libcudnn.so into the process. (do you load any other deep learning framework, or something of that sort....)

**Comment by xsacha** [comment]
Is there any issue with the same application linking multiple CUDNN? For instance, if another dependency uses CUDNN.
I notice this seems to double memory overhead but not sure if it causes any CUDNN issues.

**Comment by astooke** [comment]
Hey sorry it took me a bit to finish this...

I think this was a false alarm / coincidence.

Ran `lsof` like you said: nothing `cudnn` opens, and the only `cuda` file open outside of torch is the driver.  I re-ran with the old cudnn back in place and it learned OK, so either I had some unlucky runs earlier or I changed something else!  

Thanks for the quick responses...getting to understand PyTorch better and loving it ;)
